Invalid `.as_ptr()` type casts on some arm platforms

**Describe the bug**
When compiling mayastor with an arm compatible version of `spdk-rs` (https://github.com/openebs/spdk-rs/pull/72), mayastor still fails to build due to invalid `to_ptr()` typecasts, one example shown here:

```rust
ftl_conf.name = ftl_dev_name.as_ptr() as *mut i8;
```

On some platforms, as noted in the `CString` docs, are actually represented by `u8` and not `i8` resulting in 

```shell
error[E0308]: mismatched types
   --> io-engine/src/bdev/ftl.rs:213:25
    |
213 |         ftl_conf.name = ftl_dev_name.as_ptr() as *mut i8;
    |         -------------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*mut u8`, found `*mut i8`
    |         |
    |         expected due to the type of this binding
    |
    = note: expected raw pointer `*mut u8`
               found raw pointer `*mut i8`
```

in multiple places (including `libnvme-rs` + `io-engine` compilation)

**To Reproduce**

1. Access a `linux/aarch64` machine
2. Pull `develop` mayastor branch
3. Re-point `spdk-rs` submodule to fork at above linked issue
4. `nix-shell`
5. `cargo build`


**Expected behavior**

Full compilation!

**OS info (please complete the following information):**

Distributor ID: Ubuntu
Description:    Ubuntu 24.04.1 LTS
Release:        24.04

Mayastor: [687da6c](https://github.com/openebs/mayastor/commit/687da6c1a8d76d362e201f52f1e41cd8ae6ddca6)

--- a/io-engine/src/bdev/ftl.rs
+++ b/io-engine/src/bdev/ftl.rs
@@ -42,6 +42,7 @@ use std::{
     collections::HashMap,
     convert::TryFrom,
     fmt::{Debug, Formatter},
+    os::raw::c_char,
 };
 
 use core::ffi::c_void;
@@ -210,9 +211,9 @@ impl CreateDestroy for Ftl {
             ..unsafe { mem::zeroed() }
         };
         unsafe { spdk_ftl_get_default_conf(&mut ftl_conf, spdk_ftl_conf_size) };
-        ftl_conf.name = ftl_dev_name.as_ptr() as *mut i8;
-        ftl_conf.base_bdev = base_dev_name.as_ptr() as *mut i8;
-        ftl_conf.cache_bdev = cache_dev_name.as_ptr() as *mut i8;
+        ftl_conf.name = ftl_dev_name.as_ptr() as *mut c_char;
+        ftl_conf.base_bdev = base_dev_name.as_ptr() as *mut c_char;
+        ftl_conf.cache_bdev = cache_dev_name.as_ptr() as *mut c_char;
         ftl_conf.fast_shutdown = true;
         ftl_conf.verbose_mode = true;
         ftl_conf.mode = spdk_ftl_mode::SPDK_FTL_MODE_CREATE as u32;

--- a/io-engine/tests/memory_pool.rs
+++ b/io-engine/tests/memory_pool.rs
@@ -1,4 +1,4 @@
-use std::{collections::HashMap, ffi::CString, ptr::null_mut};
+use std::{collections::HashMap, ffi::CString, os::raw::c_char, ptr::null_mut};
 
 use once_cell::sync::OnceCell;
 
@@ -16,7 +16,7 @@ fn get_ms() -> &'static MayastorTest<'static> {
 struct TestCtx {
     id: u64,
     pos: u32,
-    ctx: *const i8,
+    ctx: *const c_char,
 }
 
 const POOL_SIZE: u64 = 128 * 1024 - 1;

--- a/libnvme-rs/src/nvme_uri.rs
+++ b/libnvme-rs/src/nvme_uri.rs
@@ -1,4 +1,4 @@
-use std::{convert::TryFrom, io, time::Duration};
+use std::{convert::TryFrom, io, os::raw::c_char, time::Duration};
 
 use url::{ParseError, Url};
 
@@ -15,14 +15,14 @@ use crate::{
 
 /// Wrapper for caller-owned C-strings from libnvme
 pub struct NvmeStringWrapper {
-    s: *mut i8,
+    s: *mut c_char,
 }
 
 impl NvmeStringWrapper {
-    pub fn new(s: *mut i8) -> Self {
+    pub fn new(s: *mut c_char) -> Self {
         NvmeStringWrapper { s }
     }
-    pub fn as_ptr(&self) -> *const i8 {
+    pub fn as_ptr(&self) -> *const c_char {
         self.s
     }
 }

